www/src/components/shell/Shell.js

```diff
@@ -37,9 +38,11 @@ function Shell({ shell }) {
   const fitAddon = useMemo(() => new FitAddon(), [])
   const [terminalTheme] = useContext(TerminalThemeContext)
   const { fresh } = useOnboarded()
+  const [restart, setRestart] = useState(false)
 
   useEffect(() => {
     if (!xterm?.current?.terminal) return
```

Comment:
change this clause to:

```js
if (!xterm?.current?.terminal || restart) {
  restart && setRestart(false)
  return
}
```

and remove the second hook.  I think the logic will be clearer that way

Reply:
oh yeah, that's definitely easier to understand :slightly_smiling_face: